src/components/WalletIcon.tsx

```diff
@@ -0,0 +1,22 @@
+import { ElementType } from 'react';
+
+import styled from 'styled-components';
+
+import { wallets } from '@/constants/wallets';
+
+import { ConnectorType, WalletInfo } from '@/lib/wallet/types';
+
+import { Icon } from './Icon';
+
+export const WalletIcon = ({ wallet, size = '1em' }: { wallet: WalletInfo; size?: string }) => {
+  if (wallet.connectorType === ConnectorType.Injected) {
+    return <StyledImage src={wallet.icon} alt={wallet.name} size={size} />;
+  }
+
+  return <Icon iconComponent={wallets[wallet.name].icon as ElementType} size={size} />;
+};
+
+const StyledImage = styled.img<{ size: string }>`
```

Comment:
super-nit: I think we are opting for `$` prefix instead of `Styled`

`StyledImage` -> `$Image`

Reply:
err mah god that is horrendous

Reply:
hahaha its mostly for the text-selection. `$Image` double tap will highlight `Image`. Whereas `StyledImage` double tap highlights `StyledImage`